diff --git a/wpiutil/CMakeLists.txt b/wpiutil/CMakeLists.txt
index eb4a0a8f7..4905f33d6 100644
--- a/wpiutil/CMakeLists.txt
+++ b/wpiutil/CMakeLists.txt
@@ -155,7 +155,7 @@ target_compile_features(wpiutil PUBLIC cxx_std_20)
 if(MSVC)
     target_compile_options(
         wpiutil
-        PUBLIC /permissive- /Zc:preprocessor /Zc:throwingNew /MP /bigobj
+        PUBLIC /permissive- /Zc:preprocessor /Zc:throwingNew /MP /utf-8 /bigobj
     )
     target_compile_definitions(wpiutil PRIVATE -D_CRT_SECURE_NO_WARNINGS)
 endif()
